errtracker/errtracker.go

```diff
@@ -57,31 +56,23 @@ var (
 	mockedHostSnapd = ""
 	mockedCoreSnapd = ""
 
-	snapConfineProfile  = "/etc/apparmor.d/usr.lib.snapd.snap-confine"
-	whoopsiePreferences = "/etc/whoopsie"
+	snapConfineProfile = "/etc/apparmor.d/usr.lib.snapd.snap-confine"
 
 	timeNow = time.Now
 )
 
 func whoopsieEnabled() bool {
-	dflt := true
-
-	cfg := goconfigparser.New()
-	err := cfg.ReadFile(whoopsiePreferences)
-	if os.IsNotExist(err) {
-		return dflt
-	}
-	if err != nil {
-		logger.Noticef("cannot read whoopsie config %q: %v", whoopsiePreferences, err)
-		return dflt
-	}
-
-	reportMetricsEnabled, err := cfg.Getbool("General", "report_metrics")
-	if err != nil {
-		logger.Noticef("cannot parse whoopsie config %q: %v", whoopsiePreferences, err)
-		return dflt
+	cmd := exec.Command("systemctl", "is-enabled", "whoopsie.service")
```

Comment:
I think it should be enough to check the exit status, which quoting the manpage is:
> Returns an exit code of 0 if at least one is enabled, non-zero otherwise.

Unless you really want to discern the case when there's no service file, then peeking at `is-enabled` output is probably ok. There are alternatives, like looking at `LoadState=(loaded|not-found)` and `SubState=(dead|<other>)` if you really want to know the exact status.

Reply:
+1 on Maciej's suggestion

Reply:
Thanks! I was considering just doing that. The downside of only the exit status is that there is no differentiation between service disabled and service not available at all. We only want to disable error report if whoopsie is installed and disabled. If it is not installed at all we proceed with error reports (unless error reports are disabled via the internal config option of course).